point/gemini_commands.py

```diff
@@ -104,26 +106,45 @@ def parse_revisions(string):
     return vals
 
 
+def parse_ip4vaddr(string):
+    match = _re_ipv4addr.fullmatch(string)
+    if match is None:                                        raise G2ResponseIPv4AddressParseError(string)
+    if int(match.group(1)) < 0 or int(match.group(1)) > 255: raise G2ResponseIPv4AddressParseError(string)
+    if int(match.group(2)) < 0 or int(match.group(2)) > 255: raise G2ResponseIPv4AddressParseError(string)
+    if int(match.group(3)) < 0 or int(match.group(3)) > 255: raise G2ResponseIPv4AddressParseError(string)
+    if int(match.group(4)) < 0 or int(match.group(4)) > 255: raise G2ResponseIPv4AddressParseError(string)
+    return ipaddress.IPv4Address(string)
+
 ####################################################################################################
 
 
-# returns tuple: (int:sign[-1|0|+1], int:deg, int:min, float:sec)
+# returns tuple: (int:sign[-1|0|+1], int:hour, int:min, int:sec)
+def ang_to_hourminsec(ang):
+    return ang_to_degminsec(ang * 24.0 / 360.0)
+
+# returns tuple: (int:sign[-1|0|+1], int:deg, int:min, int:sec)
 def ang_to_degminsec(ang):
     if   ang > 0.0: sign = +1.0
     elif ang < 0.0: sign = -1.0
     else:           sign =  0.0
     ang = abs(ang) * 3600.0
-    f_sec = ang % 60.0
+    i_sec = int(ang % 60.0) # TODO: change this to round(), if we can fix the round-up-to-60 issues
     ang /= 60.0
     i_min = int(ang % 60.0)
     ang /= 60.0
     i_deg = int(ang)
-    return (sign, i_deg, i_min, f_sec)
-# TODO: test this, in case I am dumb
+    return (sign, i_deg, i_min, i_sec)
 
-# returns tuple: (int:sign[-1|0|+1], int:hour, int:min, float:sec)
-def ang_to_hourminsec(ang):
-    return ang_to_degminsec(ang * 24.0 / 360.0)
+# returns tuple: (int:sign[-1|0|+1], int:deg, int:min)
+def ang_to_degmin(ang):
+    if   ang > 0.0: sign = +1.0
+    elif ang < 0.0: sign = -1.0
+    else:           sign =  0.0
+    ang = abs(ang) * 60.0
+    i_min = int(ang % 60.0) # TODO: change this to round(), if we can fix the round-up-to-60 issues
+    ang /= 60.0
+    i_deg = int(ang)
```

Comment:
Perhaps subjective, but seems a bit cleaner IMO to do
```
ang = abs(ang)
i_deg = int(ang)
i_min = int((ang - i_deg) * 60.0)
```

Reply:
Personally, I think it's cleaner to start at the smallest component and work linearly upward. Basically in keeping with how e.g. `ang_to_degminsec` works. I just think it reads better that way. I guess it just happens that there aren't as many steps here in `ang_to_degmin` as there are in `ang_to_degminsec`, so the laid-out step-by-step way I do it doesn't come across quite as nicely.